data_and_models/pipelines/sentence_embedding/dvc.yaml

```diff
@@ -1,4 +1,39 @@
 stages:
+  training_transformers:
+    cmd: >-
+      CUDA_VISIBLE_DEVICES=0,1,2,3
```

Comment:
```suggestion
      NGPUS=4 &&
      NGPUS_AVAIL=$(nvidia-smi --list-gpus | wc -l) &&
      if [ "$NGPUS" -gt "$NGPUS_AVAIL" ]; then echo "$NGPUS > $NGPUS_AVAIL: Not enough available GPUs!" && return 1; fi &&
      CUDA_VISIBLE_DEVICES=$(seq $NGPUS | paste -sd,)
```

Reply:
I really like the suggestion! Maybe two things: 
- It seems from the suggested changes `CUDA_VISIBLE_DEVICES=1,2,3,4`. If the environment has the exact number of GPUS available, I think that only 3 of them are going to be considered. I am not sure how to transpose this in the command line ?
- Just a thought, maybe it is completely overkilled, should we try to check memory usage of GPU before using them ?

Reply:
> I think that only 3 of them are going to be considered.

Good point! The following should work, give it a try:
```suggestion
      NGPUS=4 &&
      NGPUS_AVAIL=$(nvidia-smi --list-gpus | wc -l) &&
      if [ "$NGPUS" -gt "$NGPUS_AVAIL" ]; then echo "$NGPUS > $NGPUS_AVAIL: Not enough available GPUs!" && return 1; fi &&
      CUDA_VISIBLE_DEVICES=$(seq 0 $(($NGPUS - 1)) | paste -sd,)
```

>  should we try to check memory usage of GPU before using them ?

I also thought about that. I think it is indeed a bit overkill, especially because there is no easy way to determine if the GPU is available or not — e.g. what if it is being used at 5% in a moment, and then 95% just a millisecond after?

I just meant this small suggestion as an extra check to ensure that we are running somewhere were these GPUs _at least_ exist, but feel free to do as you whish!

Reply:
Thanks for the new suggestion, I was not sure about the syntax! I merged the suggestion in b1ee3bd.